crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -871,27 +871,40 @@ impl<'db> TypeInferenceBuilder<'db> {
         let maybe_known_class = file_to_module(self.db, body_scope.file(self.db))
             .as_ref()
             .and_then(|module| KnownClass::maybe_from_module(module, name.as_str()));
+
+        // when type_params.is_some(), there's a specialized scope for the bases + keywords
+        let bases_specialized_scope = type_params.as_ref().map(|_| {
+            self.index
+                .node_scope(NodeWithScopeRef::ClassTypeParameters(class))
+                .to_scope_id(self.db, self.file)
+        });
```

Comment:
Nit: I suggest making this a method on `ClassType`. It doesn't seem worth storing (it's very cheap to recompute). Unless we think that it helps with reducing the blast radius of file changes (e.g. so that calling `(class A).bases` where `class A` is defined in `foo.py`  from `bar.py` doesn't get invalidated when `foo.py changes). However, I do think that we might want to start caching `bases` once we add MRO resolution (which seems expensive)

Reply:
Thanks to your suggestion on using `ty` I don't even need this anymore, fortunately.